### PR TITLE
notmuch-mutt: adds missing resources

### PR DESCRIPTION
**Pull request by xsteadfastx**
In version 0.32.2 there are some perl modules missing.

- Date::Parse
- IO::Lines
- Devel::GlobalDestruction
- Sub::Exporter::Progressive
- File::Remove

- [X] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [X] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [X] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [X] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [X] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [X] Does your build pass `brew audit --strict <formula>` (after doing `brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----

### CODE REVIEW

**Comment by cho-m** [comment]
What OS are you using?
For macOS Big Sur, these are all part of system Perl and we usually don't install another copy of existing modules
```console
❯ /usr/bin/perldoc -l Date::Parse IO::Lines Devel::GlobalDestruction Sub::Exporter::Progressive File::Remove
/System/Library/Perl/Extras/5.30/Date/Parse.pm
/System/Library/Perl/Extras/5.30/IO/Lines.pm
/System/Library/Perl/Extras/5.30/Devel/GlobalDestruction.pm
/System/Library/Perl/Extras/5.30/Sub/Exporter/Progressive.pm
/System/Library/Perl/Extras/5.30/File/Remove.pm
```

If this is for macOS, then we would need to know what version and a way to test this since `brew test` is passing.

If this is for Linux, then the Perl modules need to be inside an `on_linux` section and we need to add necessary GitHub labels so CI tests Linux.

**Comment by xsteadfastx** [comment]
Hi,

On So, 08. Aug 00:29, Michael Cho wrote:
> What OS are you using?
> For macOS Big Sur, these are all part of system Perl and we usually don't install another copy of existing modules
> ```console
> ❯ /usr/bin/perldoc -l Date::Parse IO::Lines Devel::GlobalDestruction Sub::Exporter::Progressive File::Remove
> /System/Library/Perl/Extras/5.30/Date/Parse.pm
> /System/Library/Perl/Extras/5.30/IO/Lines.pm
> /System/Library/Perl/Extras/5.30/Devel/GlobalDestruction.pm
> /System/Library/Perl/Extras/5.30/Sub/Exporter/Progressive.pm
> /System/Library/Perl/Extras/5.30/File/Remove.pm
> ```

you are right. Im using a Ubuntu here. So that explains the errors.

> If this is for Linux, then the Perl modules need to be inside an `on_linux` section and we need to add necessary GitHub labels so CI tests Linux.

I added the `on_linux` to the PR. I hope this is ok.

File: Formula/notmuch-mutt.rb
Comment (cho-m) [inline Formula/notmuch-mutt.rb:7]: Currently (while merging `linuxbrew-core` and `homebrew-core`), we don't update the revision for Linux-only changes. Once PR is merged, we can bottle a copy in `homebrew-core` and a maintainer in `linuxbrew-core` can do the same there.

File: Formula/notmuch-mutt.rb
Comment (cho-m) [inline Formula/notmuch-mutt.rb:7]: Should just be removed:
```suggestion
```

**Comment by xsteadfastx** [comment]
On So, 08. Aug 13:26, Michael Cho wrote:

> Currently (while merging `linuxbrew-core` and `homebrew-core`), we don't update the revision for Linux-only changes. Once PR is merged, we can bottle a copy in `homebrew-core` and a maintainer in `linuxbrew-core` can do the same there.

Set the revision to 0. So should i do this PR on linuxbrew-core? Or is
it ok to do this here?

**Comment by cho-m** [comment]
Nothing else you need to do. The general process is right now is:
1. Merge fix into `homebrew-core`. Right now we don't really touch revision numbers here.
2. Maintainer from `linuxbrew-core` will pull fix into that repo. If necessary, they may update revision for Linux there.
3. Both repos will build bottles.

In future, Linux users will be migrated from `linuxbrew-core` to `homebrew-core` and any necessary revision alignment will be dealt with then.

**Comment by xsteadfastx** [comment]
brilliant :) thanks for the help!

**Review by SMillerDev — APPROVED** [review]

**Pull request merged**
